src/server.rs

```diff
@@ -598,8 +600,45 @@ async fn expose_metrics_and_health<H>(
         cb();
     }
 
+    #[cfg(target_os = "windows")]
+    let router = Router::new();
+
+    #[cfg(not(target_os = "windows"))]
+    let router = Router::new().route(
+        "/debug/pprof/profile",
+        get(|Query(params): Query<HashMap<String, String>>| async move {
```

Comment:
`Query` supports anything that implements `Deserialize` so maybe something something like this is nicer:

```rust
#[derive(Deserialize)]
struct Params {
    #[serde(default = "Duration::from_secs(2)")]
    seconds: Duration,
}

get(|Query(params): Query<Params>| { ... })
```

Reply:
Nice! That made things a lot cleaner. I opted for a u64, since the field is seconds, and it was not trivial to tell serde to deserialize a number into a duration. (also, apparently its not possible to use a function with parameter for the default path)